plugins/rbac-backend/src/service/plugin-endpoints.ts

```diff
@@ -38,20 +38,23 @@ export type PluginMetadataResponseSerializedRule = {
 
 export class PluginPermissionMetadataCollector {
   private readonly pluginIds: string[];
-  private urlReader: UrlReaderService;
+  private readonly urlReader: UrlReaderService;
 
   constructor(
-    private readonly discovery: PluginEndpointDiscovery,
+    private readonly discovery: DiscoveryService,
     private readonly pluginIdProvider: PluginIdProvider,
     private readonly logger: LoggerService,
     config: Config,
+    urlReader?: UrlReaderService,
```

Comment:
@PatAKnight, what do you think about using two object arguments called `deps` and `optional` instead of specifying each service individually? I generally prefer using objects for arguments when there are more than two, as it makes the function easier to read at a glance. Maybe this can be a refactor in the future?

Reply:
Yeah I agree, updating these would make the function easier to read. Definitely worth a refactor in the future.

Reply:
To keep things clean I can open up another PR to do this refactor?

Reply:
Yes, that works. Please go ahead and create a bug in JIRA, as we treat tech debt as a type of bug.

Reply:
Jira created [here](https://issues.redhat.com/browse/RHIDP-4318)